### PR TITLE
(SIMP-1695) Update pupmod for simp_options

### PR DESCRIPTION
**Pull request by jeefberkey**
- Updated assets and versions
- Converted to puppet-strings
- Switched to the new catalyst pattern with simplib::lookup
- Strong typed module

### CODE REVIEW

File: manifests/master.pp [hunk 1]
```diff
@@ -1,243 +1,132 @@
...
-  $log_to_file           = false,
-  $log_to_syslog         = true,
-  $syslog_facility       = 'LOCAL6',
-  $syslog_message_format = '%logger[%thread]: %msg',
-  $log_level             = 'WARN'
+  Simplib::IP                    $bind_address          = '0.0.0.0',
+  Simplib::IP                    $ca_bind_address       = '0.0.0.0',
+  Simplib::Port                  $ca_port               = simplib::lookup('simp_options::puppet::ca_port', { 'default_value' => 8141 }),
+  Simplib::NetList               $trusted_nets          = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1','::1'] }),
+  Pattern['^\d+y$']              $ca_ttl                = '10y',
```
Comment (trevor-vaughan) [inline manifests/master.pp]: This is not correct https://docs.puppet.com/puppet/latest/configuration.html#cattl

File: manifests/master.pp [hunk 2]
```diff
@@ -1,243 +1,132 @@
...
+  Stdlib::AbsolutePath           $codedir               = $::pupmod::params::master_config['codedir'],
+  Stdlib::AbsolutePath           $vardir                = $::pupmod::params::master_config['vardir'],
+  Stdlib::AbsolutePath           $rundir                = $::pupmod::params::master_config['rundir'],
+  Stdlib::AbsolutePath           $logdir                = $::pupmod::params::master_config['logdir'],
+  Boolean                        $use_legacy_auth_conf  = true,
+  Boolean                        $firewall              = simplib::lookup('simp_options::firewall', { 'default_value' => false }),
+  Array[Simplib::Host]           $ca_status_whitelist   = [$facts['fqdn']],
+  Optional[Stdlib::AbsolutePath] $ruby_load_path        = undef,
+  Integer                        $max_active_instances  = ($facts['processors']['count'] + 2),
+  Array                          $ssl_protocols         = ['TLSv1', 'TLSv1.1', 'TLSv1.2'],
```
Comment (trevor-vaughan) [inline manifests/master.pp]: Should be Array[String]

File: templates/etc/puppetserver/conf.d/puppetserver.conf.erb [hunk 3]
```diff
@@ -55,7 +55,7 @@ http-client: {
     # A list of acceptable protocols for making HTTP requests
     ssl-protocols: [<%= @ssl_protocols.join(',') %>]
 <% end -%>
-<% if !@ssl_cipher_suites.empty? -%>
+<% if !@ssl_cipher_suites.nil? -%>
```
Comment (trevor-vaughan) [inline templates/etc/puppetserver/conf.d/puppetserver.conf.erb]: Change to `if @ssl_cipher_suites`

File: templates/usr/local/bin/puppetagent_cron.erb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <%
-if @maxruntime.empty? then
+if @maxruntime.nil? then
```
Comment (trevor-vaughan) [inline templates/usr/local/bin/puppetagent_cron.erb]: Change to `unless @maxruntime` and drop the `then`

File: manifests/master.pp [hunk 5]
```diff
@@ -1,243 +1,132 @@
...
+  Stdlib::AbsolutePath           $vardir                = $::pupmod::params::master_config['vardir'],
+  Stdlib::AbsolutePath           $rundir                = $::pupmod::params::master_config['rundir'],
+  Stdlib::AbsolutePath           $logdir                = $::pupmod::params::master_config['logdir'],
+  Boolean                        $use_legacy_auth_conf  = true,
+  Boolean                        $firewall              = simplib::lookup('simp_options::firewall', { 'default_value' => false }),
+  Array[Simplib::Host]           $ca_status_whitelist   = [$facts['fqdn']],
+  Optional[Stdlib::AbsolutePath] $ruby_load_path        = undef,
+  Integer                        $max_active_instances  = ($facts['processors']['count'] + 2),
+  Array[String]                  $ssl_protocols         = ['TLSv1', 'TLSv1.1', 'TLSv1.2'],
+  Optional[Array]                $ssl_cipher_suites     = undef,
```
Comment (nick-markowski) [inline manifests/master.pp]: openssl_cihper_suites?

File: manifests/master.pp [hunk 6]
```diff
@@ -1,243 +1,132 @@
...
+  Boolean                        $enable_profiler       = false,
+  Array[Simplib::Hostname]       $admin_api_whitelist   = [$facts['fqdn']],
+  String                         $admin_api_mountpoint  = '/puppet-admin-api',
+  Boolean                        $log_to_file           = false,
+  Boolean                        $log_to_syslog         = true,
```
Comment (nick-markowski) [inline manifests/master.pp]: syslog catalyst

File: CHANGELOG [hunk 7]
```diff
@@ -1,4 +1,10 @@
-* Wed Nov 23 2016 Jeanne Greulich <jgreulich.simp@onyxpoint.com> - 5.0.0-0
+* Mon Dec 10 2016 Nick Miller <nick.miller@onyxpoint.com> - 7.0.0-0
```
Comment (nick-markowski) [inline CHANGELOG]: nov 21/23 was prior to release which would warrant another major bump

File: manifests/master.pp (hunk 5 above)
Comment (jeefberkey) [inline manifests/master.pp]: No I think this is for the web server, should I make that more clear?

File: manifests/master.pp [hunk 8]
```diff
@@ -120,7 +120,7 @@
   Array[Simplib::Hostname]       $admin_api_whitelist   = [$facts['fqdn']],
   String                         $admin_api_mountpoint  = '/puppet-admin-api',
   Boolean                        $log_to_file           = false,
-  Boolean                        $log_to_syslog         = true,
+  Boolean                        $syslog                = simplib::lookup('simp_options::syslog', { 'default_value' => true }),,
```
Comment (nick-markowski) [inline manifests/master.pp]: why is this defaulted to true?

File: manifests/master.pp (hunk 5 above)
Comment (nick-markowski) [inline manifests/master.pp]: eh, it's fine.

File: manifests/master.pp (hunk 8 above)
Comment (trevor-vaughan) [inline manifests/master.pp]: Agree, this should default to `false`

**Comment by jeefberkey** [comment]
👍

**Pull request merged**